cmd/kepify/main.go

```diff
@@ -0,0 +1,184 @@
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/enhancements/pkg/kepval/keps"
+)
+
+func Usage() {
+	fmt.Fprintf(os.Stderr, `
+Usage: %s [-dir <kep-directory>] [-output <path-to-json-file>]
+Command line flags override config values.
+`, os.Args[0])
+	flag.PrintDefaults()
+}
+
+func main() {
+	dirPath := flag.String("dir", "keps", "root directory for the KEPs")
+	filePath := flag.String("output", "keps.json", "output json file")
+
+	flag.Usage = Usage
+	flag.Parse()
+
+	if len(*dirPath) == 0 {
+		fmt.Fprintf(os.Stderr, "please specify the root directory for KEPs using '--dir'\n")
+		os.Exit(1)
+	}
+	if _, err := os.Stat(*dirPath); os.IsNotExist(err) {
+		fmt.Printf("directory does not exist : %s", *dirPath)
+		os.Exit(1)
+	}
+
+	if len(*filePath) == 0 {
+		fmt.Fprintf(os.Stderr, "please specify the file path for the output json using '--output'\n")
+		os.Exit(1)
+	}
+
+	// Find all the keps
+	files, err := findMarkdownFiles(dirPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to find markdown files: %v\n", err)
+		os.Exit(1)
+	}
+	if len(files) == 0 {
+		fmt.Fprintf(os.Stderr, "did not find any KEPs\n")
+		os.Exit(1)
+	}
+
+	// Parse the files
+	proposals, err := parseFiles(files)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error parsing files: %q\n", err)
+		os.Exit(1)
+	}
+
+	// Generate the json output
+	err = printJSONOutput(*filePath, proposals)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "could not open file: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func findMarkdownFiles(dirPath *string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(
+		*dirPath,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			if ignore(info.Name()) {
+				return nil
+			}
+			files = append(files, path)
+			return nil
+		},
+	)
+	return files, err
+}
+
+func parseFiles(files []string) (keps.Proposals, error) {
+	var proposals keps.Proposals
+	for _, filename := range files {
+		parser := &keps.Parser{}
+		file, err := os.Open(filename)
+		if err != nil {
+			return nil, fmt.Errorf("could not open file: %v\n", err)
+		}
+		defer file.Close()
+		kep := parser.Parse(file)
+		// if error is nil we can move on
+		if kep.Error != nil {
+			return nil, fmt.Errorf("%v has an error: %q\n", filename, kep.Error.Error())
+		}
+		fmt.Printf(">>>> parsed file successfully: %s\n", filename)
+		proposals.AddProposal(kep)
+	}
+	return proposals, nil
+}
+
+func printJSONOutput(filePath string, proposals keps.Proposals) error {
+	fmt.Printf("Output file: %s\n", filePath)
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	total := len(proposals)
+	fmt.Printf("Total KEPs: %d\n", total)
+
+	fmt.Fprintln(file, "{")
+	for i, kep := range proposals {
+		fmt.Fprintf(file, "\t\"%s\": {\n", hash(kep.OwningSIG+":"+kep.Title))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "title", kep.Title)
```

Comment:
From the example JSON I see that the titles differ in its style: like `aws-k8s-tester` vs `Custom endpoints support for AWS Cloud Provider`. Do you think we can avoid that somehow?

Reply:
not in this PR, we have to raise this as a problem in https://github.com/kubernetes/enhancements/issues/822